src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -44,7 +44,7 @@
     private final TransactionControl txControl;
     private final EntityManager em;
 
-    private List<DeviceMemberEntity> validateDeviceMembers(Long parentId, Long vsId, Long memberId, String childName) {
+    private List<DeviceMemberEntity> getDeviceMembers(Long parentId, Long vsId, Long memberId, String childName) {
```

Comment:
how does this method return a list. Since you are using the memberid which is a primary key in DeviceMemberEntity there should only be single result

Reply:
we are using single result by calling getresultlist.get(0);
getSingleResult throws exception if the query result is not found, whereas getResultList returns the empty list.
So getResultList.get(0) returns the first element and we always know that we have either one element or none.
This is agreed with Emanoel to overcome getSingleResult exception rather than catching the exception and handling it.
Please see the link which talks about the issue highlighted here 
https://stackoverflow.com/questions/2002993/jpa-getsingleresult-or-null.
Otherwise please suggest what should be the solution?

Reply:
@sudhirappaji  Arvind's point is different. Even if you must use this for a workaround you can still make this method return a single object instead of a list. Just return the first element of the result list. This way the signature if your method will be coherent.

Reply:
OK will do